pydantic/config.py

```diff
@@ -83,6 +83,7 @@ class BaseConfig:
     validate_assignment: bool = False
     error_msg_templates: Dict[str, str] = {}  # TODO remove
     arbitrary_types_allowed: bool = False  # TODO default True, or remove
+    warn_on_undefined_types: bool = True
```

Comment:
It's technically not an undefined type, rather, a non-annotated type. I guess this is pedantics (which is the whole point 😉 ), but we should choose a suitable name for this config arg.

Reply:
I think "undefined" is the best term. This warning related to annotations which are not yet defined when creating the model.

This is not about non-annotated types.

Reply:
`undefined_types_warning` would be most concise.

Reply:
Thanks for the clarification.